src/test/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensionRoutesSpec.scala

```diff
@@ -267,23 +288,35 @@ class GoogleExtensionRoutesSpec extends FlatSpec with Matchers with ScalatestRou
     when(googleIamDAO.getOrCreateServiceAccount(any[GoogleProject], any[ServiceAccountName], any[ServiceAccountDisplayName])(any[ExecutionContext])).thenReturn(Future.successful(ServiceAccount(ServiceAccountSubjectId("12312341234"), WorkbenchEmail("pet@myproject.iam.gserviceaccount.com"), ServiceAccountDisplayName(""))))
     when(googleIamDAO.createServiceAccountKey(any[GoogleProject], any[WorkbenchEmail])).thenReturn(Future.successful(ServiceAccountKey(ServiceAccountKeyId("foo"), ServiceAccountPrivateKeyData(ServiceAccountPrivateKeyData(expectedJson).encode), None, None)))
     when(googleIamDAO.removeServiceAccountKey(any[GoogleProject], any[WorkbenchEmail], any[ServiceAccountKeyId])).thenReturn(Future.successful(()))
+    when(googleIamDAO.listServiceAccountKeys(any[GoogleProject], any[WorkbenchEmail], any[Boolean])).thenReturn(Future.successful(Seq.empty))
     (googleIamDAO, expectedJson)
   }
 
+  private def createMockGoogleStorageDAOForSaKeyTests: (GoogleStorageDAO, String) = {
+    val googleStorageDAO = mock[GoogleStorageDAO]
+    val expectedJson = """{"valid":"json"}"""
+    when(googleStorageDAO.listObjectsWithPrefix(any[String], any[String])).thenReturn(Future.successful(Seq(new StorageObject().setName("foo"))))
+    when(googleStorageDAO.storeObject(any[String], any[String], any[ByteArrayInputStream], any[String])).thenReturn(Future.successful(()))
+    (googleStorageDAO, expectedJson)
```

Comment:
This function is only called in one place and that place ignores `expectedJson`.